File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task4_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdbool.h>
#include <ctype.h>
#include <string.h>

#define ALPHABET_SIZE 26

bool isIsogram(const char *str) {
    bool letters[ALPHABET_SIZE] = {false};
    
    for (int i = 0; str[i] != \'\\0\'; i++) {
        int index = tolower(str[i]) - \'a\';
        if (letters[index]) {
            return false;
        }
        letters[index] = true;
    }
    return true;
}

int main() {
    printf("%s\
", isIsogram("Dermatoglyphics") ? "true" : "false");
    printf("%s\
", isIsogram("aba") ? "true" : "false");
    printf("%s\
", isIsogram("moOse") ? "true" : "false");
    return 0;
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task4_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.HashSet;
import java.util.Set;

public class Isogram {
    public static boolean isIsogram(String str) {
        Set<Character> letters = new HashSet<>();
        for (char c : str.toLowerCase().toCharArray()) {
            if (!letters.add(c)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(isIsogram("Dermatoglyphics")); // true
        System.out.println(isIsogram("aba")); // false
        System.out.println(isIsogram("moOse")); // false
    }
}
